lang/jap/src/main/kotlin/JavaxAnnotatedImpl.kt

```diff
@@ -21,14 +21,16 @@ import com.yandex.yatagan.lang.compiled.CtAnnotationBase
 import javax.lang.model.element.AnnotationMirror
 import javax.lang.model.element.Element
 
+// See https://youtrack.jetbrains.com/issue/KT-51087, which was fixed in [1, 9, 0] metadata version.
```

Comment:
Target version in KT-51087 is 1.8.20-Beta. Is this version already have 1.9.0 metadata version?

Reply:
Well, It seemed to me, but I better double check

Reply:
Turns out, metadata version in the code, generated by 1.8.20 compiler, is still `[1, 8, 0]`, so such check is not sound.

Moreover, if the `kotlin.Metadata` is present, it doesn't mean the code is from KAPT (a stub), it may well be an already compiled bytecode, which always had correct annotation order. And there's no way to determine whether the element is backed by source or bytecode.

So, the task to make a sound decision here seems impossible :(